storage/src/global_state/state/mod.rs

```diff
@@ -505,6 +506,78 @@ pub trait CommitProvider: StateProvider {
             )),
         }
     }
+
+    /// Forcibly unbonds delegator bids which fall outside configured delegation limits.
+    fn forced_undelegate(&self, request: ForcedUndelegateRequest) -> ForcedUndelegateResult {
```

Comment:
The only place i see that calls this is a test (`should_forcibly_undelegate_after_setting_validator_limits`).

Reply:
You are right, it seems something got messed up when resolving conflicts. I restored the call within `execute_finalized_block`.
I'm still not entirely certain this is the best place to handle pre-auction cleanup, but this was the compromise for now. Do you think it would be better to place it for example somewhere in `commit_step`?